lrlex/src/lib/lexer.rs

```diff
@@ -213,16 +215,13 @@ impl<'a, StorageT: Copy + Eq> LRLexer<'a, StorageT> {
 impl<'a, StorageT: Copy + Eq + Hash + PrimInt + Unsigned> Lexer<StorageT>
     for LRLexer<'a, StorageT>
 {
-    /// Return all this lexer's lexemes or a `LexError` if there was a problem when lexing.
-    fn lexemes(&mut self) -> Result<Vec<Lexeme<StorageT>>, LexError> {
-        let mut i = 0; // byte index into s
-        let mut lxs = Vec::new(); // lexemes
-
-        while i < self.s.len() {
+    fn next(&mut self) -> Option<Result<Lexeme<StorageT>, LexError>> {
+        while self.i < self.s.len() {
+            let old_i = self.i;
```

Comment:
Why do we need `old_i`? It seems to me that within this context, `old_i` is always equal to `self.i`.

Reply:
At line 257, we increment `self.i` but still want `old_i`. I could rewrite it not to do this though, if you think that would be clearer. [I'm neutral on this.]

Reply:
Ah I missed that. Should we maybe just do `let old_i = self.i` just before line 257?

Reply:
Ah, looking again, we'd also need to do this in lines 246 & 250. So it might be better off as it is.

Reply:
You are right. In that case it's better to leave it as is.